samples/csharp_dotnetcore/50.teams-messaging-extensions-search/Bots/TeamsMessagingExtensionsSearchBot.cs

```diff
@@ -99,5 +136,153 @@ private async Task<IEnumerable<(string, string, string, string, string)>> FindPa
+                Content = adaptiveList.Content,
+                Preview = previewcard.ToAttachment()
+            };
+
+            // return attachment;
+
+            return new MessagingExtensionResponse
+            {
+                ComposeExtension = new MessagingExtensionResult
+                {
+                    Type = "result",
+                    AttachmentLayout = "list",
+                    Attachments = new List<MessagingExtensionAttachment> { attachment }
+                }
+            };
+        }
+
+
+        public MessagingExtensionResponse GetConnectorCard()
+        {
+          
+            string filepath = "./connectorCard.json";
+
+            var json = File.ReadAllText(filepath);
+
+            var cardJson = JsonConvert.DeserializeObject<ConnectorJsonSerializer>(json);
+
+            cardJson.sections[0].activityImage = _configuration+"/imgConnector.jpg";
+
+            var ConnectorCardJson = JsonConvert.SerializeObject(cardJson);
+
+            var previewcard = new ThumbnailCard
+            {
+                Title = "O365 Connector Card",
+                Text = "Please select to get Connector card"
+            };
+
+            var connector = FetchConnector(ConnectorCardJson);
+            var attachment = new MessagingExtensionAttachment
+            {
+                ContentType = O365ConnectorCard.ContentType,
+                Content = connector.Content,
+                Preview = previewcard.ToAttachment()
+            };
+
+            return new MessagingExtensionResponse
+            {
+                ComposeExtension = new MessagingExtensionResult
+                {
+                    Type = "result",
+                    AttachmentLayout = "list",
+                    Attachments = new List<MessagingExtensionAttachment> { attachment }
+                }
+            };
+        }
+
+
+        public static Attachment FetchAdaptive(string filepath)
+        {
+            var adaptiveCardJson = File.ReadAllText(filepath);
+            var adaptiveCardAttachment = new Attachment
+            {
+                ContentType = /*"application/vnd.microsoft.card.adaptive"*/ AdaptiveCard.ContentType,
+                Content = JsonConvert.DeserializeObject(adaptiveCardJson),
+            };
+
+            return adaptiveCardAttachment;
+
+        }
+
+
+        public Attachment FetchConnector(string cardJson)
+        {
+
+           // var connectorCardJson = File.ReadAllText(filepath);
+            var connectorCardAttachment = new MessagingExtensionAttachment
+            {
+                ContentType = O365ConnectorCard.ContentType/*"application/vnd.microsoft.teams.card.o365connector"*/,
+                Content = JsonConvert.DeserializeObject(cardJson),
+
+            };
+
+            return connectorCardAttachment;
+        }
+
+
+        public MessagingExtensionResponse GetResultGrid()
+        {
+            var files = Directory.GetFiles("wwwroot");
```

Comment:
Please try the search pattern parameter, then it will not be necessary to loop through the files checking names with a regex.